### ISSUE TITLE
examples - unable to import generated classes

### ISSUE THREAD

**Issue by paddymahoney** [ferraith/setuptools-antlr#21]
Hello ferraith! 

Thanks for your work here. I'm attempting to incorporate this into a setuptools project. I'm finding that I am unable to access the generated classes using an import statement, whereas projects outside of ./build/lib appear to be accessible. 

I seem to be able to reproduce using the example project - after activating a venv, I do 'python setup.py antlr' then 'python setup.py build' followed by 'python setup.py install' . I then install ipython (pip install ipython), and attempt to do variations of 'from foobar.dsl.bar.Bar import *' to no avail. 

Any suggestions here? otherwise, the code generation under ./build/lib is as expected.

**Comment by ferraith** [comment]
Hi Patrick!

I think the root cause of your problem is your configured PYTHONPATH. Typically in IDEs like PyCharm the build folder isn't part of the PYTHONPATH. The default output folder build\lib does perfectly fit if your are interested in building a python package including your generated parser without polluting  your project directory. Because during packaging (e.g. python setup.py bdist_wheel) everything goes into build\lib it also makes sense to generate the parsers into the build\lib folder.

Nevertheless in your developer scenario I would prefer a different output folder. Therefore you have the option to control the output folder with two option e.g. in your setup.cfg:
```
#output = [default=<output path>]
 #              [<root-level grammar>=<output path> ...]
...
# Output goes into -o directories regardless of paths/package (yes|no); default: no
#x-exact-output-dir = yes
```

For the foobar example you maybe configure it like the following:
```
output = default=.
```
All the output will be generated into the root folder of your project:
```
.\foobar\dsl\bar
.\foobar\dsl\foo
```

Of course you can also use the x-exact-output-dir option to skip generation of subfolders.

**Comment by paddymahoney** [comment]
thanks! would you accept a pr that changes the default to ./ ?

**Comment by ferraith** [comment]
When I started working on this setuptools command I already pondered if I should generate the code into the source or in the build tree. There are reasons for both.
Source Tree:
+ A developer can directly import the generated parser without the need to modify the pythonpath
+ It's more like a user of the tool would expect it using setuptools-antlr the first time
- The source tree is polluted with generated code
- python setup.py clean --all doesn't work anymore because it only deletes content in the build folder - I haven't a solution for that apart from implementing a --clean option.
- During code-generation setuptools-antlr have to check the whole output path if a __init__.py file exists in each folder. If not a __init__.py have to be generated. This is because the build command (python setup.py build) only copies code which are placed in packages.
- *.tokens and *.interp aren't shipped any more with the wheel package. This isn't really a problem because the user is likely not interested in these files.

I will create a feature branch and will work in the next days on that change. If we are happy with the new behavior I will push it into the master and will build a new release 0.2.0.

**Comment by ferraith** [comment]
I think we are done here. I shipped a new release 0.2.0 which generates the parsers in the package source directory. Additionally I implemented a logic to ensure the generated code will be transfered to the build directory when the `build` command is executed.
